google/gax/__init__.py

```diff
@@ -42,5 +42,79 @@ class PageDescriptor(
             ['request_page_token_field',
              'response_page_token_field',
              'resource_field'])):
-    """Describes the structure of a page-streaming call"""
+    """Describes the structure of a page-streaming call."""
     pass
+
+
+class BundleDescriptor(
+        collections.namedtuple(
+            'BundleDescriptor',
+            ['bundled_field',
+             'request_descriminator_fields',
+             'response_field'])):
+    """Describes the structure of bundled call.
+
+    request_descriminator_fields may include '.' as a separator, which is used
+    to indicate object traversal.  This allows fields in nested objects to be
+    used to determine what requests to bundle.
+
+    Attributes:
+      bundled_field: the repeated field in the request message that
+        will have it messages aggregated by bundling
+      request_descriminator_fields: a set fields in the
+        target request message class that are used to determine
+        which messages should be bundled together.
+      response_field: the optional repeated field in the response
+        that contains distinct values that per each distinct message
+        in the bundle field
+
+    """
+    pass
+
+
+class BundleOptions(
+        collections.namedtuple(
+            'BundleOptions',
+            ['message_count_threshold',
+             'message_bytesize_threshold',
+             'delay_threshold'])):
+    """Holds values used to configure bundling.
+
+    The xxx_threshold attributes are used to configure when the bundled request
+    should be made.
+
+    Attributes:
+        message_count_threshold: the bundled request will be sent once
+            the count of outstanding messages reaches this value
+        message_bytesize_threshold: the bundled request will be sent once
+            the count of bytes in the outstanding messages reaches this value
+        delay_threshold: the bundled request will be sent this amount of
+            time after the first message in the bundle was added to it.
+
+    """
+    # pylint: disable=too-few-public-methods
+
+    def __new__(cls,
+                message_count_threshold=1,
```

Comment:
This means that if you just set delay_threshold to a positive value, it has no effect, since the message_count_threshold (which defaults to 1) always kicks in first.

Instead, let's default all values to 0 and have an assertion that there is some positive, nonzero threshold in the constructor.